packages/core/saved-objects/core-saved-objects-base-server-internal/src/model_version/build_transform_fn.test.ts

```diff
@@ -9,119 +9,318 @@
 import { loggerMock } from '@kbn/logging-mocks';
 import type {
   SavedObjectsModelChange,
-  SavedObjectModelTransformationFn,
   SavedObjectModelTransformationDoc,
   SavedObjectModelTransformationContext,
+  SavedObjectModelDataBackfillFn,
 } from '@kbn/core-saved-objects-server';
 import { buildModelVersionTransformFn } from './build_transform_fn';
 
 describe('buildModelVersionTransformFn', () => {
-  const stubTransform = (): jest.MockedFn<SavedObjectModelTransformationFn> =>
-    jest.fn().mockImplementation((doc: SavedObjectModelTransformationDoc) => ({ document: doc }));
+  const stubDatabackfill = (): jest.MockedFn<SavedObjectModelDataBackfillFn> =>
+    jest.fn().mockImplementation((doc: SavedObjectModelTransformationDoc) => ({}));
 
   const createContext = (): SavedObjectModelTransformationContext => ({
     log: loggerMock.create(),
     modelVersion: 42,
   });
 
-  const createDoc = (): SavedObjectModelTransformationDoc => ({
+  const createDoc = <T = any>(attributes: T = {} as T): SavedObjectModelTransformationDoc<T> => ({
     id: 'foo',
     type: 'bar',
-    attributes: {},
+    attributes,
   });
 
-  it('builds the transform from a single function', () => {
-    const transform1 = stubTransform();
-    const changes: SavedObjectsModelChange[] = [{ type: 'data_backfill', transform: transform1 }];
+  it('returns a no-op function when no transform functions are present', () => {
```

Comment:
We we log transforms progress and result during migrations or somewhere else? Being about to monitor a saved object model progression would help with debugging and failure analysis.

Reply:
Transform failures / stacktraces will be shown in the logs, as we were doing for the old migration function. 

If by outcome you mean the transformed document, then no it doesn't appear in the logs (and we can't, for obvious security reasons)